src/ipips/ipip-0428.md

```diff
@@ -0,0 +1,164 @@
+   V2-only records that are protobuf with only two fields: Data
+   CBOR+signatureV2.
+
+### User benefit
+
+- End users: the main benefit for end user is the smaller size of IPNS Records and
+  less complexity during creation/validation of modern V2-only records.
+
+- Developers interested in IPNS: by making IPNS Record creation as simple as
+  "create DAG-CBOR with these fields, and sign it", and validation to
+  "signatureV2 should match the DAG-CBOR value and key". We've removed surface
+  for bugs, making it easier to reason about for use in greenfield projects.
+
+- IPFS ecosystem: lowering the complexity related to IPNS record creation and
+  validation makes it more likely for third-party interoperable IPNS
+  implementations to happen.
+
+### Compatibility
+
+- This is backward-compatible, we adjust validation logic to allow V2-only
+  records, but all V1+V2 records that are being used in the wild today are
+  still valid
+
+- V2-only rollout is not part of this IPIP.
+  - Our suggestion is to default to creating V1+V2 records for now, keeping
+    backward-compatibility with the old IPNS clients.
+
+  - Creation of V2-only records should be introduced as an explicit opt-in. It
+    is up to implementations to decide when it is feasible to default to
+    creating V2-only records on IPNS publish.
+
+### Security
+
+- `IpnsEntry.signatureV1` (protobuf field) is parsed only by legacy clients, modern ones ignore this value
+
+It is highly advised to implement validation conformance tests using the fixtures
+included at the end of this IPIP.
+
+### Alternatives
+
+Describe alternate designs that were considered and related work.
+
+1. Just switch to V2-only as the new default!
+   - No, this would be a breaking change. We have to do this in two steps,
+     because we've rushed the way V2 was introduced in 2021, and STILL require
+     field copying, even when `signatureV1` is missing. So technically there
+     was never "V2", it was more like "V1.5". Only with this IPIP, we finally
+     adjust validation to only care about CBOR values when there is no
+     `signatureV1`.
+
+2. Why keeping the outer protobuf envelope? Could we make IPNS DAG-CBOR-only?
+   - Due to how long it takes for decentralized network nodes to upgrade, we prefer evolution rather than revolution.
+   - Protobuf is a useful envelope for two reasons:
+     1. Ensures the opaque V2-only record can be passed and stored in existing infrastructure.
+     2. Allows us to evolve IPNS record ("V3") in the future without impacting existing infrastructure.
+
+## Test fixtures
+
+TODO: IPNS record that is valid for 100 years and
+
+1. V1-only → record invalid
+1. V1+V2 (both signatures valid) → record valid
+1. V1+V2 (both signatures valid, but 'value' is different in V1 pb) → record invalid
+1. V1+V2 (only signatureV1 valid) → record invalid
+1. V1+V2 (only signatureV2 valid) → record valid
+1. V2-only (no V1 fields) → record valid
```

Comment:
@hacdias I've filled https://github.com/ipfs/gateway-conformance/issues/144 would you be able to do create these fixtures and related gateway-conformance tests in next 2 weeks? (before next IPFS Implementers call)

Reply:
@lidel https://github.com/ipfs/gateway-conformance/pull/157

Reply:
Thanks, I've updated the IPIP to referece test vectors from your PR :+1: